nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/AbstractTestAnnotationNotification.java

```diff
@@ -153,23 +159,52 @@ protected void captureAndVerifyAnnotationEventNotification(Supplier<DocumentMode
             assertEquals(1, handledEvents.size());
             Event expectedEvent = handledEvents.get(0);
             assertEquals(annotationEventType, expectedEvent.getName());
-            checkDocumentEventContext(expectedEvent, annotationDocumentModel, annotatedDocumentModel);
-            checkReceivedMail(emailsResult.getMails(), annotationDocumentModel, annotatedDocumentModel,
-                    handledEvents.get(0), annotationEventType);
+
+            checkDocumentEventContext(expectedEvent, annotationDocumentModel, annotationParentDocumentModel,
+                    annotatedDocumentModel);
+
+            checkReceivedMail(emailsResult.getMails(), annotationDocumentModel, annotatedDocumentModel, handledEvents.get(0),
+                    annotationEventType);
+            return annotationDocumentModel;
         }
     }
 
+    @Test
+    public void shouldNotifyWithTheRightAnnotatedDocument() {
+        // First comment
+        Annotation createdAnnotation = createAnnotation(annotatedDocumentModel);
+        DocumentModel createdAnnotationDocModel = session.getDocument(new IdRef(createdAnnotation.getId()));
+        // before subscribing, or previous event will be notified as well
+        transactionalFeature.nextTransaction();
+        // Reply
+        captureAndVerifyAnnotationEventNotification(() -> {
+            // subscribe to notifications
+            NuxeoPrincipal principal = session.getPrincipal();
+            String subscriber = NotificationConstants.USER_PREFIX + principal.getName();
+            String notif = "CommentAdded";
+            notificationService.addSubscription(subscriber, notif, annotatedDocumentModel, false, principal, notif);
+
+            Comment reply = createAnnotation(createdAnnotationDocModel);
+            DocumentModel replyDocumentModel = session.getDocument(new IdRef(reply.getId()));
+            return session.getDocument(new IdRef(replyDocumentModel.getId()));
+        }, COMMENT_ADDED, DOCUMENT_CREATED);
+    }
```

Comment:
Wasn't the ticket about having the wrong document in the notification for the *first comment*? Here you are only validating the document for a reply. You are probably missing another test for the first comment/annotation?

Reply:
The first part of the problem was solved by @RSalem07  in previous ticket: [NXP 28253 Fix wrong notification sent on comment edit](https://jira.nuxeo.com/browse/NXP-28253)
It introduced a new API in this [commit](https://github.com/nuxeo/nuxeo/commit/b3f5a8838af3709c9123823ac18aacd526e7d349#diff-d6ec9b58ab4af25556e2a6754c50ea22R361)
supported by `PropertyCommentManager` and `TreeCommentManager`
Tested in:
[AbstractTestCommentNotification#shouldNotifyEventWhenUpdateComment](https://github.com/nuxeo/nuxeo/commit/b3f5a8838af3709c9123823ac18aacd526e7d349#diff-b5ba4c2a86feb19034eeccb42bbcd8aaR113)
[AbstractTestAnnotationNotification#shouldNotifyEventWhenUpdateAnnotation](https://github.com/nuxeo/nuxeo/commit/b3f5a8838af3709c9123823ac18aacd526e7d349#diff-28554eaa66542e2d29a24a2ced0d0134R110)

I'll add that to the [28252 ticket](https://jira.nuxeo.com/browse/NXP-28252)